nomad/structs/config/users.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package config
+
+import (
+	"errors"
+
+	"github.com/hashicorp/nomad/helper/pointer"
+)
+
+// UsersConfig configures things related to operating system users.
+type UsersConfig struct {
+	// MinDynamicUser is the lowest uid/gid for use in the dynamic users pool.
+	MinDynamicUser *int `hcl:"dynamic_user_min"`
+
+	// MaxDynamicUser is the highest uid/gid for use in the dynamic users pool.
+	MaxDynamicUser *int `hcl:"dynamic_user_max"`
```

Comment:
```suggestion
	// MinDynamicUser is the lowest uid/gid for use in the dynamic users pool.
	MinDynamicUser *int `hcl:"min_guid"`

	// MaxDynamicUser is the highest uid/gid for use in the dynamic users pool.
	MaxDynamicUser *int `hcl:"max_guid"`
```

Minor suggestion, but since this config is already scoped in an `users` block, I think it's safe to drop the `user` component? `dynamic` also didn't seem to convey much information to me, and it wasn't very clear what the value represents (is it a count? a group ID? etc.) but I may be wrong 😬 

Before:

```hcl
users {
  dynamic_user_min = 10000
  dynamic_user_max = 20000
}
```

After:

```hcl
users {
  min_guid = 10000
  max_guid = 20000
}
```

I would maybe suggest even making `users` more explicit, like `task_users`, but I don't know if that's a good name.

Reply:
The problem with `users.min_guid` etc. is that this isn't a reference to generic users on the system - it's specific to the new concept of "Dynamic Workload Users" which is exposed for task drivers who expressly indicate support through a new task driver Capability. 

Even more so, a generic name would be conflicting with the very similar sounding restrictions @mikenomitch is working on in https://github.com/hashicorp/nomad/pull/20073